src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFsDriver.Linux.cs

```diff
@@ -78,6 +78,8 @@ public SysFsDriver()
         /// <param name="pinNumber">The pin number in the driver's logical numbering scheme.</param>
         protected internal override void OpenPin(int pinNumber)
         {
+            const int numOpenRetries = 100; // wait 1 second max for udev to set the permissions
```

Comment:
Can we instead define max time this will wait? Thread.Sleep is not really good way to measure time and I think this will sometimes take 1s and other time 15s or more.

(Check permissions immediately and then if you need to wait then create a stopwatch and check time once in a while).

Reply:
OK I have done this. On my PI 3b it takes only 1 retry so < 10ms for udev to do its thing PR updated

@joperezr @krwq 

I seem to be having repeated problems with the automated checks where everything other than the build release passes.

Reply:
yeah, if you follow the link of the failed build into Azure DevOps, and click on the Tests Tab you'll see that a test failed. We run our tests in actual hardware which causes sometimes for the tests to be a bit flaky, which we are planning to fix. I have restarted the build in the meantime.